docs/modules/ROOT/pages/how-tos/configuring/labels-and-annotations.adoc

```diff
@@ -29,16 +29,47 @@ tasks:
       value:
         - from-pull-request=true
         - hermetic-param=$(params.hermetic)
+    - name: ANNOTATIONS <.>
+      value:
+        - my-annotation=true
+        - foo=bar
 ...
 ----
 
-<.> The LABELS param accepts an array of labels to be applied to the image after it is built.
+<.> The LABELS and ANNOTATIONS params accepts an array of labels and annotations to be applied to the image after it is built.
+
+[[generating-dynamic-labels-or-annotations]]
+== Generating dynamic labels or annotations
 
-[[generating-dynamic-labels]]
-== Generating dynamic labels
+You can use the *generate-labels* task to produce some dynamic labels or annotations. To do so, add the generate-labels task to the list of tasks in your pipeline, e.g.:
 
-You can use the *generate-labels* task to produce some dynamic labels. To do so, add the generate-labels task to the list of tasks in your pipeline, e.g.:
+[source,yaml]
+----
+tasks:
+...
+    - name: generate-labels
+      params:
+        - name: label-templates
+          value: 
+             - "build-date=$SOURCE_DATE" <.>
+             - "short-commit=$(tasks.clone-repository.results.short-commit)"
+        - name: source-date-epoch
+          value: '$(tasks.clone-repository.results.commit-timestamp)'
+      runAfter:
+        - clone-repository
+      taskRef:
+        params:
+          - name: name
+            value: generate-labels
+          - name: bundle
+            value: quay.io/konflux-ci/tekton-catalog/task-generate-labels:0.1@sha256:0068fe8b3c5d010daee5a922781a74cfb82251e775c260d14d9e50dd1a7aca65
+          - name: kind
+            value: task
+        resolver: bundles
+...
+----
 
+You can add the task twice to generate dynamic labels and annotations:
```

Comment:
Can you not just use one invocation and reapply the same properties on both annotations and labels (this is what you describe in the `combining approaches` section at the end)?

Or are you saying that you can invoke the `generate-labels` task to create separate sets of dynamic annotations and labels?

Reply:
oh, that's a great point. Yeah, my initial thought was that I would need to use the generate-labels twice to get labels and annotations, but you are right, you can create the full list once and then pass some values to one param and some others to the other. Let me update the doc for this :)